File: crates/analyzer/general/tests/corpus/bad/taint_flow.go
// EXPECT: gen_command_injection
// Injection reached across several lines — no sink line contains a source.
package main

func ping(w http.ResponseWriter, r *http.Request) {
	host := r.FormValue("host")
	args := "ping -c 1 " + host
	exec.Command("sh", "-c", args).Run()
}

File: crates/analyzer/general/tests/corpus/bad/taint_flow.py
# EXPECT: gen_sql_injection
# EXPECT: gen_command_injection
"""Injection reached across several lines — no sink line contains a source."""
from flask import request


def lookup():
    uid = request.args["id"]
    query = "SELECT * FROM users WHERE id = " + uid
    cur.execute(query)


def ping():
    host = request.form["host"]
    cmd = "ping -c 1 " + host
    os.system(cmd)
